cmd/kubelet/app/server.go

```diff
@@ -1080,6 +1081,16 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 			if q.Sign() == -1 {
 				return nil, fmt.Errorf("resource quantity for %q cannot be negative: %v", k, v)
 			}
+
+			if v1.ResourceName(k) == v1.ResourceMemory {
+				valueStr := q.String()
+				lastChar := valueStr[len(valueStr)-1:]
+				if lastChar != "i" {
```

Comment:
This doesn't seem like the right check. Shouldn't we be more concerned with the actual value specified and whether it meets the minimum required?

Reply:
So m is different with M?

Reply:
> So m is different with M?

Yeah, different. For memory, `100m` will be parsed as `"100 milli"`, instead of `"100 M"`.

Reply:
two comments:

1. m vs. M, two different meanings by same letter different case; that's easy to be confused.
1. memory also support integer, can user argue `100m` means `100 milli byte`?